library/ssl_tls.c

```diff
@@ -2610,6 +2618,13 @@ static int ssl_tls13_session_load(mbedtls_ssl_session *session,
     session->max_early_data_size = MBEDTLS_GET_UINT32_BE(p, 0);
     p += 4;
 #endif
+#if defined(MBEDTLS_SSL_RECORD_SIZE_LIMIT)
+    if (end - p < 2) {
+        return MBEDTLS_ERR_SSL_BAD_INPUT_DATA;
```

Comment:
I'd like to think that if someone was using Mbed TLS just before this PR, and had a saved session, then recompiled with this PR, and tried to resume, they'd get an `MBEDTLS_ERR_SSL_VERSION_MISMATCH` (considering it the version of the data, rather than just the version of the library) error, not a `MBEDTLS_ERR_SSL_BAD_INPUT_DATA` error, although I suspect the caller would treat both identically.

That would suggest that `MBEDTLS_SSL_RECORD_SIZE_LIMIT` should be incorporated in `SSL_SERIALIZED_SESSION_CONFIG_BITFLAG`

However, the same reasoning applies to `MBEDTLS_SSL_EARLY_DATA`, so this looks like pre-existing, rather than anything else, and I'm mainly making the observation here, rather than asking for a change.

Reply:
> That would suggest that `MBEDTLS_SSL_RECORD_SIZE_LIMIT` should be incorporated in `SSL_SERIALIZED_SESSION_CONFIG_BITFLAG`

That's a good point. I think we overlooked this aspect of serialized sessions while working on new session tickets, early data and now record size limit. I am creating an issue to address that: #8686.